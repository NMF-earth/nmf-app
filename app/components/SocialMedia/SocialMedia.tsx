import React from "react";
import { View, TouchableOpacity } from "react-native";
import { MaterialCommunityIcons } from "@expo/vector-icons";
import { Linking } from "expo";

import { Colors } from "../../style";

import styles from "./SocialMedia.styles";

const socialMedia = [
  {
    iconName: "slack",
    url:
      "https://join.slack.com/t/not-my-fault/shared_invite/enQtODMzMzcyNjExMzc3LTVlNDQwMTA0OGUxMmJmNTE4MTA4MTgwZGZmNTBiZmRhNTcxY2FlN2MzZGQyMmRjZDVjN2Q2MjFlNzE4ZGI5ZTI",
  },
  {
    iconName: "github-circle",
    url: "https://github.com/NMF-earth",
  },
  {
    iconName: "twitter",
    url: "https://twitter.com/nmf_earth",
  },
  {
    iconName: "mastodon",
    url: "https://mastodon.social/@nmf_earth",
  },
  {
    iconName: "medium",
    url: "https://medium.com/@nmf_earth",
  },
];

export default () => (
  <View style={styles.container}>
    <View style={styles.box}>
      {socialMedia.map((item, index) => (
        <TouchableOpacity key={index} onPress={() => Linking.openURL(item.url)}>
          <MaterialCommunityIcons
            name={item.iconName}
            size={32}
            color={Colors.grey70}
          />
        </TouchableOpacity>
      ))}
    </View>
  </View>
);
