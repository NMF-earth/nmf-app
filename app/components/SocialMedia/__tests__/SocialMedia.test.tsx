import React from "react";
import { create } from "react-test-renderer";
import { Linking } from "expo";

import SocialMedia from "../";

jest.unmock("../");

beforeEach(() => {
  Linking.openURL = jest.fn();
});

test("renders correctly SocialMedia", () => {
  const tree = create(<SocialMedia />).toJSON();
  expect(tree).toMatchSnapshot();
});

test("should open link", () => {
  const root = create(<SocialMedia />).root;
  // eslint-disable-next-line @typescript-eslint/ban-ts-ignore
  // @ts-ignore
  const icons = root.findAllByType("TouchableOpacity");
  icons.map((item) => item.props.onPress());
  expect(Linking.openURL).toHaveBeenCalledTimes(icons.length);
});
