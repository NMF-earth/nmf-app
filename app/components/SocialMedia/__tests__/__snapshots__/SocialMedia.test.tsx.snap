// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly SocialMedia 1`] = `
<View
  style={
    {
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      {
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "maxWidth": 200,
        "paddingBottom": 50,
      }
    }
  >
    <TouchableOpacity
      onPress={[Function]}
    >
      <View
        color="#595D58"
        name="github"
        size={32}
      />
    </TouchableOpacity>
    <TouchableOpacity
      onPress={[Function]}
    >
      <View
        color="#595D58"
        name="mastodon"
        size={32}
      />
    </TouchableOpacity>
    <TouchableOpacity
      onPress={[Function]}
    >
      <View
        color="#595D58"
        name="medium-m"
        size={32}
      />
    </TouchableOpacity>
  </View>
</View>
`;
